File: CS/DXSample.Module/TestObject.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.ExpressApp.Editors;

namespace DXSample.Module {
    [DefaultClassOptions]
    public class TestObject : BaseObject {
        public TestObject(Session session) : base(session) { }
        private string _Error;
        [RuleRequiredField(ResultType = ValidationResultType.Error)]
        public string Error {
            get { return _Error; }
            set { SetPropertyValue("Error", ref _Error, value); }
        }
        private string _Warning;
        [RuleRequiredField(ResultType = ValidationResultType.Warning)]
        public string Warning {
            get { return _Warning; }
            set { SetPropertyValue("Warning", ref _Warning, value); }
        }
        private string _Information;
        [RuleRequiredField(ResultType = ValidationResultType.Information)]
        public string Information {
            get { return _Information; }
            set { SetPropertyValue("Information", ref _Information, value); }
        }
    }
}

File: VB/DXSample.Module/TestObject.vb
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation
Imports DevExpress.ExpressApp.Editors

Namespace DXSample.Module
	<DefaultClassOptions>
	Public Class TestObject
		Inherits BaseObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Error As String
		<RuleRequiredField(ResultType := ValidationResultType.Error)>
		Public Property [Error]() As String
			Get
				Return _Error
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Error", _Error, value)
			End Set
		End Property
		Private _Warning As String
		<RuleRequiredField(ResultType := ValidationResultType.Warning)>
		Public Property Warning() As String
			Get
				Return _Warning
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Warning", _Warning, value)
			End Set
		End Property
		Private _Information As String
		<RuleRequiredField(ResultType := ValidationResultType.Information)>
		Public Property Information() As String
			Get
				Return _Information
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Information", _Information, value)
			End Set
		End Property
	End Class
End Namespace
